Sources/VitaminUIKit/Components/Snackbar/VitaminSnackbar.swift

```diff
@@ -0,0 +1,360 @@
+            titleLabel.textAlignment = .left
+            messageLabel.textAlignment = .left
+            let margins = rightMargin + imageWidthAndHeight + imageLeftAndRightMargin
+            messageLabel.preferredMaxLayoutWidth = maxWidth - margins
+        } else {
+            NSLayoutConstraint.activate(withoutImageConstraints)
+            NSLayoutConstraint.deactivate(withImageConstraints)
+            icon.removeFromSuperview()
+            titleLabel.textAlignment = .center
+            messageLabel.textAlignment = .center
+            messageLabel.preferredMaxLayoutWidth = maxWidth - 2 * rightMargin
+        }
+    }
+
+    // launch or invalidate timer for aut dismissal when the autodismiss prperty is set
+    private func applyNewAutoDismiss(_ newValue: Bool) {
+        if autoDismiss, !newValue {
+            timer?.invalidate()
+        } else if !autoDismiss, newValue {
+            launchAutoDismissal()
+        }
+    }
+
+    // apply changes when dismiss delay is set
+    private func applyNewDismissDelay() {
+        timer?.invalidate()
+        if autoDismiss {
+            launchAutoDismissal()
+        }
+    }
+
+    // apply changes when dismissOnTap property is set
+    private func applyNewDismissOnTap() {
+        if let gestureRecognizer = gestureRecognizer {
+            if dismissOnTap {
+                self.addGestureRecognizer(gestureRecognizer)
+            } else {
+                self.removeGestureRecognizer(gestureRecognizer)
+            }
+        }
+    }
+
+    // apply layout changes when maxWidth property is set
+    private func applyNewMaxWidth() {
+        NSLayoutConstraint.deactivate(commonConstraints)
+        commonConstraints = []
+        setupCommonConstraints()
+        NSLayoutConstraint.activate(commonConstraints)
+    }
+}
+
+// MARK: Private methods for dismissal or display handling
+extension VitaminSnackbar {
+    // launch the timer that will automatically dismiss
+    private func launchAutoDismissal() {
+        timer = Timer.scheduledTimer(withTimeInterval: TimeInterval(dismissDelay), repeats: false) { _ in
+            self.internalDismiss()
+        }
+    }
+
+    // internal method to dismiss the `VitaminSnackbar`
+    private func internalDismiss() {
+        UIView.animate(
+            withDuration: VitaminSnackbar.appearAndDisappearDuration,
+            delay: 0
+        ) {
+            self.titleLabel.layer.opacity = 0
+            self.messageLabel.layer.opacity = 0
+            self.icon.layer.opacity = 0
+            self.layer.opacity = 0
+        } completion: { _ in
+            self.removeFromSuperview()
+        }
+    }
+
+    /// Make the snackbar appear by fading it in
+    /// This method should only be called before first display of the snackbar, becaise it will force opacity to 0.
+    /// If the `VitaminSnackbar`is already displayed when you call this method, it will first instantly disappear, then fade in
```

Comment:
```suggestion
    /// This method should only be called before first display of the snackbar, because it will force opacity to 0.
    /// If the `VitaminSnackbar` is already displayed when you call this method, it will first instantly disappear, then fade in
```